File: 347-TopKFrequentElements/347-TopKFrequentElements.cpp
// Last updated: 2/19/2026, 1:35:04 PM
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        // Count frequencies using a map
        map<int, int> myMap;
        for (int x : nums) {
            ++myMap[x];
        }

        // Store frequencies and their corresponding numbers in a vector
        vector<pair<int, int>> myVec;
        for (auto [num, freq] : myMap) {
            myVec.emplace_back(freq, num);
        }

        // Sort the vector by frequency in descending order
        sort(myVec.begin(), myVec.end(), greater<pair<int, int>>());

        // Retrieve the top-k elements
        vector<int> ans;
        for (int i = 0; i < k; i++) {
            ans.push_back(myVec[i].second);
        }

        return ans;
    }
};

File: 347-TopKFrequentElements/347-TopKFrequentElements.py
# Last updated: 5/29/2026, 1:42:45 AM
1class Solution:
2    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
3        freq = {}
4        for num in nums:
5            if num in freq:
6                freq[num]+=1
7            else:
8                freq[num] = 1
9            
10        max_heap = [(-value, key) for key, value in freq.items()]
11        heapq.heapify(max_heap)
12
13        answer = []
14        for i in range(k):
15            item = heapq.heappop(max_heap)
16            answer.append(item[1])
17
18        return answer
19